plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -332,8 +332,14 @@ else if (identity.getId() == TRINO_MERGE_PARTITION_DATA) {
                     }
                 });
 
+        TupleDomain<IcebergColumnHandle> dynamicFilterTupleDomain = dynamicFilter.getCurrentPredicate().transformKeys(IcebergColumnHandle.class::cast);
+        if (dynamicFilterTupleDomain.getDomains().isPresent() && !dynamicFilterTupleDomain.isAll() && !partitionKeys.isEmpty()) {
```

Comment:
Are any of these conditions actually necessary ? It seems to me that the below code should work even if any these are not true.

Reply:
> Are any of these conditions actually necessary ?

Probably not. My reasoning was to avoid unnecessary computation when the table is not partitioned.